centaur/src/it/scala/centaur/AbstractCentaurTestCaseSpec.scala

```diff
@@ -44,14 +45,17 @@ abstract class AbstractCentaurTestCaseSpec(cromwellBackends: List[String], cromw
   def executeStandardTest(testCase: CentaurTestCase): Unit = {
     def nameTest = s"${testCase.testFormat.testSpecString} ${testCase.workflow.testName}"
 
-    def runTest(): IO[SubmitResponse] = testCase.testFunction.run
+    def runTestAndDeleteZippedImports(): IO[SubmitResponse] = for {
+      submitResponse <- testCase.testFunction.run
+      _ = cleanUpImports(testCase.workflow.data) // cleanup imports in the end of test
+    } yield submitResponse
 
     // Make tags, but enforce lowercase:
     val tags = (testCase.testOptions.tags :+ testCase.workflow.testName :+ testCase.testFormat.name) map { x => Tag(x.toLowerCase) }
     val isIgnored = testCase.isIgnored(cromwellBackends)
     val retries = if (testCase.workflow.retryTestFailures) ErrorReporters.retryAttempts else 0
 
-    runOrDont(nameTest, tags, isIgnored, retries, runTest())
+    runOrDont(nameTest, tags, isIgnored, retries, runTestAndDeleteZippedImports())
```

Comment:
Hmmm.... I may be missing something, but I _think_ we should remove the `()` - so that the function is only called if `runOrDont` actually decides to run the test?

Reply:
Not necessarily. `runOrDont` still accepts function, not the result.

Reply:
hmm... I see that `runOrDont` would **accept** either, but by including the `()` it looks to me like it's being given the result rather than being given the function? ie:
* `runTestAndDeleteZippedImports` has type `() => IO[...]`
* `runTestAndDeleteZippedImports()` has type `IO[...]`

Reply:
Actually, you are right 🙂. I forgot that `runTest: => IO[SubmitResponse]` in the function declarations means that it's a by-name parameter, which will only be evaluated when used. So it looks like the logic here is still correct - we pass result of the function call into the `tryTryAgain`, but it will only be evaluated when accessed within the `tryTryAgain` function.